docs/decisions/0013-backend-for-frontend-api.md

```diff
@@ -0,0 +1,69 @@
+  and helper function `queryEnterpriseLearnerDashboardBFF` ([source](https://github.com/openedx/frontend-app-learner-portal-enterprise/blob/337a7b44d94d8be5d9233e80e2fa0e2de72d165c/src/components/app/data/queries/queries.ts#L270))
+- `useBFF` ([source](https://github.com/openedx/frontend-app-learner-portal-enterprise/blob/337a7b44d94d8be5d9233e80e2fa0e2de72d165c/src/components/app/data/hooks/useBFF.js#L16))
+  - Since the introduction of React Query and custom hooks, the migration process involved abstracting a generalized `useBFF` hook
+    that determines the following conditionals:
+    - Within `resolveBFFQuery` ([source](https://github.com/openedx/frontend-app-learner-portal-enterprise/blob/337a7b44d94d8be5d9233e80e2fa0e2de72d165c/src/components/app/data/queries/utils.js#L11))
+      - Whether a customer is eligible to use the BFF API call by either being explicitly enabled via the configuration API as a defined `enterpriseUuid`,
+        or an individual customer has been selected to resolve the dashboard page with the BFF loader with a gradual rollout process via a Waffle flag.
+      - Whether the current route the user is on is eligible to make the BFF call, if so, return the query required to call the BFF.
+    - For a matched route where a BFF call can be made, but the user is ineligible to make the call, we fall back to the default query provided by a
+      an argument
+      - `useSubscriptions` ([source](https://github.com/openedx/frontend-app-learner-portal-enterprise/blob/337a7b44d94d8be5d9233e80e2fa0e2de72d165c/src/components/app/data/hooks/useSubscriptions.js#L11))
+        - This is the simplest implementation of migrating existing React query custom hooks to utilize the BFF. It passes an object representing 
+          fields typically passed within `useQuery` for the BFF query call within `bffQueryConfig`. This includes the transform to match the
+          data structure of the existing `useSubscriptions` non-BFF call to `useBFF` within an object when making a call to the BFF query. It also passes
+          the original arguments for making a query call within `useSubscriptions` passed within `useQuery` as a `fallbackQueryConfig` object within `useBFF`.
+        - This structure allows the developer to continue using the original hooks as implemented.
+      - `useEnterpriseCourseEnrollments` ([source](https://github.com/openedx/frontend-app-learner-portal-enterprise/blob/337a7b44d94d8be5d9233e80e2fa0e2de72d165c/src/components/app/data/hooks/useEnterpriseCourseEnrollments.js#L36))
+        - The modifications required to make this current hook utilize the original `useEnterpriseCourseEnrollments` hook and
+          the `useBFF` layer required specific options to be passed in depending on the modified hooks downstream related to `enterpriseCourseEnrollments`.
```

Comment:
These hooks appear to be siblings of `resolveBFFQuery`, as a child under `useBFF`, despite these being their own hooks that _consume_ `useBFF`. Should these should be siblings to `useBFF` instead, as they're not part of `useBFF` as `resolveBFFQuery` is?

Perhaps there's a new group as sibling of `useBFF` with the label of something like "Custom query hooks" and a sentence or two description pertaining to "supporting the incremental migration to the BFF layer without impacting existing query hooks to retrieve data throughout the application".

Then, the specifics about `useSubscriptions` and `useEnterpriseCourseEnrollments` could be child items under that parent "Custom query hooks"?